src/Raven.Client/Documents/Operations/AI/AiAgent/StartChatOperation.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Json.Serialization;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.AI.AiAgent;
+
+internal class StartChatOperation<T> : IMaintenanceOperation<ChatResult<T>> where T : new()
+{
+    private readonly string _agent;
+    private readonly string _prompt;
+    private readonly Dictionary<string, string> _parameters;
+
+    public StartChatOperation(string agent, string prompt, Dictionary<string, string> parameters = null)
```

Comment:
```suggestion
    public StartChatOperation(string agent, string prompt, Dictionary<string, object> parameters = null)
```

I think we need this as an object - or a JSON object, instead

Reply:
I can change that, but I think we need to have a broader discussion on how exactly we are working with the params